### PR TITLE
executor, codec: fix bug when join 2 decimal keys.

### PR DESCRIPTION
**Pull request by hanfei1991**
a supplyment for #3784

### CODE REVIEW

**Comment by hanfei1991** [comment]
@XuHuaiyu PTAL

File: util/codec/codec.go [hunk 1]
```diff
@@ -79,7 +79,17 @@ func encode(b []byte, vals []types.Datum, comparable bool, hash bool) ([]byte, e
 			b = EncodeInt(b, int64(val.GetMysqlDuration().Duration))
 		case types.KindMysqlDecimal:
 			b = append(b, decimalFlag)
-			b = EncodeDecimal(b, val)
+			if hash {
+				dec := val.GetMysqlDecimal()
+				precision, frac := dec.PrecisionAndFrac()
```
Comment (shenli) [inline util/codec/codec.go]: Add comment here.

File: util/codec/codec.go [hunk 2]
```diff
@@ -79,7 +79,19 @@ func encode(b []byte, vals []types.Datum, comparable bool, hash bool) ([]byte, e
 			b = EncodeInt(b, int64(val.GetMysqlDuration().Duration))
 		case types.KindMysqlDecimal:
 			b = append(b, decimalFlag)
-			b = EncodeDecimal(b, val)
+			if hash {
```
Comment (XuHuaiyu) [inline util/codec/codec.go]: add a comment for `hash`.

Comment (hanfei1991) [inline util/codec/codec.go]: Have added in the beginning of this function

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**